### ISSUE TITLE
Facebook popup blocked

### ISSUE THREAD

**Issue by awwester** [Vestorly/torii#210]
In both Chrome and Firefox our Facebook popup is being blocked using torii 0.5.1. We also use Google authentication and that is working fine. Recently we have upgraded torii from 0.3.X to 0.5.1 and before the upgrade the popup was not being blocked. After the first popup is blocked, the second one goes through and authenticates normally.

I saw https://github.com/Vestorly/torii/issues/183 but unfortunately that didn't solve the problem, or maybe I misunderstood but I changed `this.get('torii').open` to `this.get('session').open` but that is undefined.

Here is my controller code:

      authenticateSocial: function(provider){
      var flashMessages = this.get('flashMessages');
      var endpoint = provider.match('facebook|twitter|google')[0];

      flashMessages.clearMessages();

      this.get('torii').open(provider).then((authData) => {
        var data = {};
        data[endpoint+'Login'] = { 'access_token': authData.accessToken || authData.authorizationToken.access_token }

        Ember.$.ajax(this.paths.auth(endpoint), {
          data: JSON.stringify(data),
          method: 'post',
          contentType: 'application/json'
        }).then((data) => {
          this.get('session').authenticate('authenticator:rlg', data[endpoint+'Login'].key);
        }, function(error){
          flashMessages.danger('There was an error logging you in.')
        });

      }, function(error){
        console.log(error);
        flashMessages.danger('There was an error logging you in.')
      });
    },

Any ideas?

**Comment by minasmart** [comment]
I'm running in to the same issue.

Looking at facebook's [docs](https://developers.facebook.com/docs/reference/javascript/FB.login/v2.4) I see this:

> Calling FB.login() results in the JS SDK attempting to open a popup window. As such, this method should only be called after a user click event, otherwise the popup window will be blocked by most browsers.

I'm wondering if this is getting caused by user event transforming into a deferred action based on the sdk being loaded.

Also, of note: [diffing the facebook-connect provider between 0.3.5 and master](https://gist.github.com/minasmart/6595cf68feba14b005c3) only shows the `return_scopes` option being added, and `grantScopes` being added to the normalized response; so this may have been a problem for a while?

**Comment by minasmart** [comment]
Getting close to a fix. It has to do with the eager facebook loading not working, which can be traced back to this commit: https://github.com/Vestorly/torii/commit/029ba58ac2fd683a55b2179973dccc64ae96524c

For whatever reason, the `walk-providers` code isn't being run.

**Comment by minasmart** [comment]
@awwester If you're using the current (ish) ember cli and ember, with torii as an add-on, there seems to be an issue where Torii's initializers aren't being called.

If you add the following to your `app.js` somewhere:
```javascript
import toriiLoadInitializers from 'torii/load-initializers'; 

...

toriiLoadInitializers(App, config.modulePrefix);
```

**Comment by tmayr** [comment]
Hey @minasmart , I'm working with @awwester on the same project, just tired your workaround, and it indeed works, hopefully a proper fill will come out soon, but in the meantime, this will do just fine.

Thanks for sharing!

**Comment by sbounmy** [comment]
having the same issue here. Tried @minasmart fix but now error raised :
```Uncaught TypeError: Application.instanceInitializer is not a function``` 

Figured out that ```Application.instanceInitializer``` have been introduced on Ember 1.12+... as I am running on 1.8.1 I have no escape :(

**Comment by BillBrower** [comment]
I'm having trouble importing `toriiLoadInitializers` with version 0.8.2. When  I do I get this error in the console.

`Could not find module 'torii/initializers/initialize-torii' imported from 'torii/load-initializers'.`

`torii/initializers/initialize-torii` seems to be in my node_modules folder though.

<img width="297" alt="screen shot 2017-01-23 at 6 32 41 pm" src="https://cloud.githubusercontent.com/assets/1717845/22227697/88cb8396-e19a-11e6-90c8-bb4fd7e5ce82.png">

Have any of you experienced this problem?

**Comment by v3ss0n** [comment]
Facebook popup being block at latest again.

**Comment by drewcovi** [comment]
@minasmart any idea what would be happening on the latest version of ember? running into this as well
